system/lib/fetch/wasmfs.cpp

```diff
@@ -0,0 +1,55 @@
+// // Copyright 2021 The Emscripten Authors.  All rights reserved.
+// // Emscripten is available under two separate licenses, the MIT license and the
+// // University of Illinois/NCSA Open Source License.  Both these licenses can be
+// // found in the LICENSE file.
+
+#include <stdlib.h>
+#define __NEED_struct_iovec
+#include <emscripten/emscripten.h>
+#include <wasi/api.h>
+
+extern "C" {
+
+  __wasi_errno_t __wasi_fd_write( __wasi_fd_t fd, const __wasi_ciovec_t *iovs, size_t iovs_len, __wasi_size_t *nwritten) {
+    if (fd == 1 || fd == 2) {
+      MAIN_THREAD_EM_ASM({
+        var buffer = [];
+        var fd = $0;
+        var iovs = $1;
+        var iovs_len = $2;
+        var nwritten = $3;
+        var num = 0;
+        for (var i = 0; i < iovs_len; i++) {
+          var ptr = HEAP32[(((iovs)+(i*8))>>2)];
```

Comment:
```suggestion
          var ptr = HEAP32[iovs + i*8 >> 2];
```
The JS order of operations allows this to be written without parens.